hardcode zone as default-zone

critterjohnson opened the pull request:
Removes `zone` from `inputs.cue`, and moves it to `gm/intermediates.cue` and `k8s/intermediates.cue`. `k8s/outputs/catalog.cue` and `k8s/outputs/controlensemble.cue` both reference `mesh.spec.zone`, and have been changed as well.

## Test

Can be tested with

```
cue eval -c ./gm/outputs --out text -e mesh_configs_yaml
cue eval -c ./k8s/outputs --out text -e k8s_manifests_yaml
```

critterjohnson commented:
fixes [sc-24466]

tilleryd commented:
Please add a changelog entry/

tilleryd commented:
@critterjohnson Have you tested this by installing on k8s? I created a new branch from yours named `remove-zone-tillery-test`, merged main into it, and deployed to Rancher. The operator is failing health checks.

```
Readiness probe failed: Get "http://10.42.0.9:8081/readyz": dial tcp 10.42.0.9:8081: connect: connection refused
```

critterjohnson commented:
Will reproduce and see what's going on. Interesting that it can eval / export from CUE but fails in the operator.

critterjohnson commented:
Added a changelog entry and fixed the issue @tilleryd was seeing.

Removing the `zone` field from `inputs.cue` made the CUE un-exportable, which the operator code relies on. To fix, I added cue.mod/gen/github.com/greymatter-io/common/api/meshv1/mesh_types.cue to provide a sensible default on the zone object. I went with this over making it an optional field for a couple of reasons:
- we probably shouldn't imply that `zone` is actually optional
- the value would have to be changed in `gm/intermediates.cue`, several files in `gm/outputs`, and a couple places in `k8s/intermediates.cue`

This is the CUE standard for giving generated code defaults per [their docs](https://cuelang.org/docs/integrations/go/#extract-cue-from-go) (see "Mixing manually created with generated files"), but we have not done it before, so it would definitely be good to get some eyes on it. I am happy to just make it a default field and change it in a few different places.

danielpcox commented:
I hope someone is actually going to deploy a mesh and confirm it works with this change? `cue eval` is frequently not enough of a test. I'd also suggest at least turning Spire on, because that opens more code paths.

critterjohnson commented:
Ah - I did deploy an actual mesh and confirmed that it works.

danielpcox commented:
Thanks Critter, you are the man.

tilleryd commented on cue.mod/gen/github.com/greymatter-io/common/api/meshv1/mesh_types.cue:
```diff
@@ -0,0 +1,5 @@
+package meshv1
```
Please add a comment at the top explaining the purpose of this file. Specifically that it is intended to override types in `mesh_types_go_gen.cue`.

tilleryd commented:
I deployed successfully (without Spire) as a cursory check. I like the minimal changes with the final iteration to make this happen.  Nice job.
